### PR TITLE
volume_group resource

### PR DESCRIPTION
**Pull request by alanwthatcher**
### Description

This initial submission will provide basic functionality of mkvg and chvg.  If this code is desirable for the the cookbook, I can commit info to README or make any other changes as required.
### Issues Resolved

[List any existing issues this PR resolves]

### Check List
- [ ] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @alanwthatcher

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by alanwthatcher** [comment]
Made some changes based on browsing some other code here

**Comment by chmod666org** [comment]
@alanwthatcher Please sign the Chef CLA. Modify readme.md with provider documentation. Will review the rest of the code tomorrow.

**Comment by alanwthatcher** [comment]
CLA is signed.  I have not had much chance to work on it today.  I am working on the idempotency for the :change action.

**Comment by alanwthatcher** [comment]
Added revised code and modified readme.md

File: providers/volume_group.rb [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  # Get unused disk array
+  unused_disks = unused_disk_search
+  Chef::Log.info("unused disks: #{unused_disks}")
+  # Interate through array to find size of disk, keep if larger than size requested
+  unless unused_disks.nil?
+    disk_choices = {}
+    unused_disks.each do |this_disk|
+      disk_name = this_disk
+      Chef::Log.info("mkvg: examining disk size for #{this_disk}")
+      bootinfo = Mixlib::ShellOut.new("bootinfo -s #{this_disk}")
```
Comment (chmod666org) [inline providers/volume_group.rb]: Use getconf /dev/#{this_disk} here (bootinfo is deprecated)

File: providers/volume_group.rb [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    unless disk_chosen.nil?
+      disk = disk_chosen[0]
+    end
+  end
+  disk
+end
+
+def unused_disk_search
+  # Find unused disk and put into array
+  unused_disk = Mixlib::ShellOut.new("lspv | awk '{if ($3 == \"None\") print $1;}' |  tr '\n' ' '")
```
Comment (chmod666org) [inline providers/volume_group.rb]: Avoid multiple pipes in ShellOut,. Maybe you can do the same thing with a single pipe lspv | awk '$3 == "None" {printf "%s ",$1}'

File: README.md [hunk 3]
```diff
@@ -627,14 +627,72 @@ Actions:
...
+* `major_number` (optional) - add - Specify major number to be used at creation
+* `mirror_pool_strictness` (optional) - add - Enable mirror pool strictness
+* `mirror_pool` (optional) - add - Specify mirror pool
+* `infinite_retry` (optional) - add - Enables the infinite retry option of the logical volume.
+* `non_concurrent_varyon` (optional) - add - Volume group not allowed to varyon in non-concurrent mode in more than one node at the same time.
+* `critical_vg` (optional) - add - Eneable critical VG option
+* `auto_synchronize` (optional) - change - Sets the synchronization characteristics for the volume group
+* `hotspare` (optional) - change - Sets the sparing characteristics for the volume group
+* `lost_quorom_varyoff` (optional) - change - Determines if the volume group is automatically varied off after losing its quorum of physical volumes.
+
```
Comment (chmod666org) [inline README.md]: Explain disk, best_fit and use_all_disks parameters here.

**Comment by chmod666org** [comment]
@alanwthatcher Added some comments in your code. Tell me what you think of this. Then I'll merge the request

**Comment by alanwthatcher** [comment]
Updated code based on suggestions (thanks!), and readme.md updated also.

**Pull request merged**